common/src/main/java/org/apache/falcon/security/DefaultAuthorizationProvider.java

```diff
@@ -170,7 +171,8 @@ public void authorizeResource(String resource, String action,
                     authorizeAdminResource(authenticatedUGI, action);
                 }
             } else if ("entities".equals(resource) || "instance".equals(resource)) {
-                authorizeEntityResource(authenticatedUGI, entityName, entityType, action);
+                boolean entityTypeOptional = "entities".equals(resource) && LIST_OPERATION.equals(action);
```

Comment:
Good point, @pallavi-rao . I moved the "skip" check outside the authorizeEntityResource method.